File: implementations/java-springboot/harness/test/testdata/utc-timestamp-source/good-elapsed-time-outside-scope/common/web/RequestLoggingInterceptor.java
package com.example.accountservice.common.web;

import java.time.LocalDateTime;

// Outside the rule's scope (neither domain/ nor persistence/) — a stopwatch reading is
// location-independent and must not be flagged.
public class RequestLoggingInterceptor {
    public long elapsedSeconds(LocalDateTime start) {
        return java.time.Duration.between(start, LocalDateTime.now()).toSeconds();
    }
}

File: implementations/kotlin-springboot/harness/test/testdata/utc-timestamp-source/good/src/main/kotlin/common/RequestLoggingInterceptor.kt
package com.example.accountservice.common

import java.time.LocalDateTime

// Measures how long the request took. The reading is never stored, only subtracted from a later
// one, so the zone it resolves against is irrelevant — this file is outside the rule's scope and
// produces no finding at all.
class RequestLoggingInterceptor {
    fun start(): LocalDateTime = LocalDateTime.now()
}
